tests/unit/ProcessMaker/InboxRules/ApplyActionTest.php

```diff
@@ -0,0 +1,132 @@
+        ]);
+
+        $activeTask = ProcessRequestToken::factory()->create([
+            'user_id' => $user->id,
+            'status' => 'ACTIVE',
+            'element_id' => 'UserTaskUID',
+        ]);
+
+        MatchingTasks::shouldReceive('matchingInboxRules')
+            ->once()
+            ->with($activeTask)
+            ->andReturn([$inboxRule]);
+
+        WorkflowManager::shouldReceive('completeTask')
+            ->once()
+            ->with(
+                $activeTask->process,
+                $activeTask->processRequest,
+                $activeTask,
+                ['foo' => 'bar']
+            );
+
+        ApplyAction::applyActionOnTask($activeTask);
+    }
+
+    public function testMarkAsPriority()
+    {
+        $user = User::factory()->create();
+
+        $taskForInboxRule = ProcessRequestToken::factory()->create([
+            'user_id' => $user->id,
+            'status' => 'COMPLETED',
+        ]);
+
+        $inboxRule = InboxRule::factory()->create([
+            'process_request_token_id' => $taskForInboxRule->id,
+            'user_id' => $user->id,
+            'mark_as_priority' => true,
+            'submit_data' => null,
+        ]);
+
+        $activeTask = ProcessRequestToken::factory()->create([
+            'user_id' => $user->id,
+            'status' => 'ACTIVE',
+            'element_id' => 'UserTaskUID',
+            'is_priority' => false,
+        ]);
+
+        MatchingTasks::shouldReceive('matchingInboxRules')
+            ->once()
+            ->with($activeTask)
+            ->andReturn([$inboxRule]);
+
+        ApplyAction::applyActionOnTask($activeTask);
+
+        $activeTask->refresh();
+
+        $this->assertTrue($activeTask->is_priority);
+    }
+
+    public function testReassignToUserID()
+    {
+        $user = User::factory()->create([
+            'is_administrator' => true,
+            'status' => 'ACTIVE'
+        ]);
+        $this->actingAs($user);
+
+        $userToReassign = User::factory()->create();
+
+        $taskForInboxRule = ProcessRequestToken::factory()->create([
+            'user_id' => $user->id,
+            'status' => 'ACTIVE',
+        ]);
+
+        $inboxRule = InboxRule::factory()->create([
+            'process_request_token_id' => $taskForInboxRule->id,
+            'user_id' => $user->id,
+            'reassign_to_user_id' => $userToReassign->id,
+            'submit_data' => null,
+        ]);
+
+        $activeTask = ProcessRequestToken::factory()->create([
+            'user_id' => $user->id,
+            'status' => 'ACTIVE',
+            'element_id' => 'UserTaskUID',
+        ]);
+
+        MatchingTasks::shouldReceive('matchingInboxRules')
+            ->once()
+            ->with($activeTask)
+            ->andReturn([$inboxRule]);
+
+        ApplyAction::applyActionOnTask($activeTask);
+        
+        $activeTask->refresh();
+        
+        $this->assertEquals($userToReassign->id,$activeTask->user_id);
```

Comment:
So it looks like `reassignTo` in ProcessMaker/Models/ProcessRequestToken.php kicks off a process to re-assign the task. So for the test, you can `WorkflowManager::shouldReceive('runProcess')->once()->with(.....) See the method in ProcessMaker/Models/ProcessRequestToken.php for the correct values. `->andReturn(12345)` any number

It also sets `$this->user_id = $res['assign_to'];` but does not save it. So remove `refresh()` and assert it equals `12345`